src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -391,6 +395,15 @@ int dhcp_server_send_packet(sd_dhcp_server *server,
         if (r < 0)
                 return r;
 
+        if (req->agent_info_option) {
+                size_t opt_full_length = *(req->agent_info_option + 1) + 2;
+                /* there must be space left for SD_DHCP_OPTION_END also */
+                if (optoffset + opt_full_length + 1 <= req->max_optlen) {
```

Comment:
Not a matter, but maybe `optoffset + oot_full_length < req->max_optlen` is simpler?